components/lib/nearby/src/main/java/mozilla/components/lib/nearby/NearbyConnection.kt

```diff
@@ -0,0 +1,342 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.nearby
+
+import android.Manifest
+import android.content.Context
+import android.os.Build
+import com.google.android.gms.nearby.Nearby
+import com.google.android.gms.nearby.connection.AdvertisingOptions
+import com.google.android.gms.nearby.connection.ConnectionInfo
+import com.google.android.gms.nearby.connection.ConnectionLifecycleCallback
+import com.google.android.gms.nearby.connection.ConnectionResolution
+import com.google.android.gms.nearby.connection.ConnectionsClient
+import com.google.android.gms.nearby.connection.DiscoveredEndpointInfo
+import com.google.android.gms.nearby.connection.DiscoveryOptions
+import com.google.android.gms.nearby.connection.EndpointDiscoveryCallback
+import com.google.android.gms.nearby.connection.Payload
+import com.google.android.gms.nearby.connection.PayloadCallback
+import com.google.android.gms.nearby.connection.PayloadTransferUpdate
+import com.google.android.gms.nearby.connection.PayloadTransferUpdate.Status
+import com.google.android.gms.nearby.connection.Strategy
+import mozilla.components.support.base.log.logger.Logger
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.concurrent.ConcurrentHashMap
+import mozilla.components.support.base.observer.Observable
+import mozilla.components.support.base.observer.ObserverRegistry
+
+/**
+ * A class that can be run on two devices to allow them to connect. This supports sending a single
+ * message at a time in each direction. It contains internal synchronization and may be accessed
+ * from any thread
+ *
+ * @constructor Constructs a new connection, which will call [NearbyConnectionListener.updateState]
+ *     with an argument of type [ConnectionState.Isolated]. No further action will be taken unless
+ *     other methods are called by the client.
+ * @param context context needed to initiate connection, used only at start
+ * @param name name shown by this device to other devices
+ */
+class NearbyConnection(
+    private val context: Context,
+    private val observer: NearbyConnectionObserver,
+    private val name: String = Build.MODEL
+) : Observable<NearbyConnectionObserver> by ObserverRegistry() {
```

Comment:
Kotlin tip! If you use `by observer` and give `observer` a default ObserverRegistry(), you can then use that single instance and register multiple observers.

See the [LegacySessionManager](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/session/src/main/java/mozilla/components/browser/session/LegacySessionManager.kt#L24).

Reply:
Thank you. I have removed the observer from the constructor parameter list.